.github/workflows/CI.yml

```diff
@@ -44,6 +44,23 @@ jobs:
       - uses: gap-actions/process-coverage@v2
       - uses: codecov/codecov-action@v2
 
+  test-cygwin:
+    name: "master cygwin64 HPC no"
+    if: ${{ !(github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) }}
+    runs-on: windows-latest
+    env:
+      CHERE_INVOKING: 1
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gap-actions/setup-cygwin@v1
+      - uses: gap-actions/setup-gap@cygwin-v2
```

Comment:
Is there a reason to keep the cygwin versions of those script apart, or can simply fold this into the v2 tag of setup-gap (and similar for the others) ?

Reply:
Huh, in fact there doesn't seem to be a `cygwin-v2` tag on that repository? see https://github.com/gap-actions/setup-gap/tags

Reply:
In principle, these scripts should be unified. The only difference between `setup-gap@cygwin-v2` and `setup-gap@v2` is that they use different `shell:` configurations in their steps.

But I have not yet found a way of being able to doing this.

Reply:
The GitHub Actions composite actions do not seem to be well developed. Only a month ago (!) did it become possible to include `if:` configurations in the steps of a GitHub Actions! I imagine it will be a while until it is possible to dynamically configure the `shell:` configuration of the steps.

Reply:
i.e. ideally we want this structure of `gap-actions/setup-gap`:
- first determine what the operating system is, and set a variable `CUSTOM_SHELL` (either `bash` for Mac/Linux, or `C:\cygwin64\bin\bash.exe`)
- then, on each step, including the configuration `shell: ${{ env.CUSTOM_SHELL }}`.

However, at least the last time I checked, dynamically setting the shell, i.e. doing `shell: ${{ env.CUSTOM_SHELL }}`, has not yet been implemented for composite GitHub Actions.

If this was possible, then two Cygwin and non-Cygwin versions could very easily be unified.

Reply:
I see, thanks. Very annoying. Hrm

Reply:
I am hoping that this will eventually become possible. For now, it is not a huge pain to keep a separate branch in the `gap-actions` repositories for the Cygwin versions - the diffs are very small (basically just the `shell:` lines).

Reply:
> Huh, in fact there doesn't seem to be a `cygwin-v2` tag on that repository? see https://github.com/gap-actions/setup-gap/tags

Sorry @fingolfin I missed this. For now `cygwin-v2` is a branch https://github.com/gap-actions/setup-gap/tree/cygwin-v2 - I've done the same thing with the other repositories too.

Reply:
ahhhhh of course! stupid me, I didn't even consider the possibility of a branch, that explains why I could find it 😂